plugins/modules/aci_interface_policy_leaf_fc_policy_group.py

```diff
@@ -0,0 +1,361 @@
+                "name": "production"
+            }
+        }
+    }
+filter_string:
+  description: The filter string used for the request
+  returned: failure or debug
+  type: str
+  sample: ?rsp-prop-include=config-only
+method:
+  description: The HTTP method used for the request to the APIC
+  returned: failure or debug
+  type: str
+  sample: POST
+response:
+  description: The HTTP response from the APIC
+  returned: failure or debug
+  type: str
+  sample: OK (30 bytes)
+status:
+  description: The HTTP status from the APIC
+  returned: failure or debug
+  type: int
+  sample: 200
+url:
+  description: The HTTP url used for the request to the APIC
+  returned: failure or debug
+  type: str
+  sample: https://10.11.12.13/api/mo/uni/tn-production.json
+"""
+
+from ansible.module_utils.basic import AnsibleModule
+from ansible_collections.cisco.aci.plugins.module_utils.aci import ACIModule, aci_argument_spec, aci_annotation_spec, aci_owner_spec
+
+
+def main():
+    argument_spec = aci_argument_spec()
+    argument_spec.update(aci_annotation_spec())
+    argument_spec.update(aci_owner_spec())
+    argument_spec.update(
+        # NOTE: Since this module needs to include both infra:FcAccPortGrp (for FC) and infra:FcAccBndlGrp (for FC PC):
+        # NOTE: The user(s) can make a choice between (port(FC), port_channel(FC PC))
+        lag_type=dict(type="str", required=True, aliases=["lag_type_name"], choices=["port", "port_channel"]),
+        policy_group=dict(type="str", aliases=["name", "policy_group_name"]),  # Not required for querying all objects
+        description=dict(type="str", aliases=["descr"]),
+        fibre_channel_interface_policy=dict(type="str", aliases=["fibre_channel_interface_policy_name"]),
+        port_channel_policy=dict(type="str", aliases=["port_channel_policy_name"]),
+        attached_entity_profile=dict(type="str", aliases=["aep_name", "aep"]),
+        state=dict(type="str", default="present", choices=["absent", "present", "query"]),
+        name_alias=dict(type="str"),
+    )
+
+    module = AnsibleModule(
+        argument_spec=argument_spec,
+        supports_check_mode=True,
+        required_if=[
+            ["state", "absent", ["policy_group"]],
+            ["state", "present", ["policy_group"]],
+        ],
+    )
+
+    policy_group = module.params.get("policy_group")
+    description = module.params.get("description")
+    lag_type = module.params.get("lag_type")
+    fibre_channel_interface_policy = module.params.get("fibre_channel_interface_policy")
+    port_channel_policy = module.params.get("port_channel_policy")
+    attached_entity_profile = module.params.get("attached_entity_profile")
+    state = module.params.get("state")
+    name_alias = module.params.get("name_alias")
+
+    aci = ACIModule(module)
+
+    if lag_type == "port":
+        aci_class_name = "infraFcAccPortGrp"
+        dn_name = "fcaccportgrp"
+    elif lag_type == "port_channel":
```

Comment:
Will never be something else, so we could use else

```suggestion
    else lag_type == "port_channel":
```

Reply:
Mapping could also be used to set values in a dict then the conditional is not needed but you can refer to the map

Reply:
reesolving as discussed